c/tskit/tables.h

```diff
@@ -2558,7 +2555,42 @@ int tsk_table_collection_build_index(tsk_table_collection_t *self, tsk_flags_t o
 @brief Runs integrity checks on this table collection.
 
 @rst
-TODO: document me. https://github.com/tskit-dev/tskit/issues/592
+
+Checks the integrity of this table collection. The default checks (i.e., with
+options = 0) guarantee the integrity of memory and references within the
+table collection. All spatial values (along the genome) are checked
+to see if they are finite values and within the required bounds. Time values
+are checked to see if they are finite or marked as unknown.
+
+To check if a set of tables
+is a valid tree sequence, use the TSK_CHECK_ALL option. More fine-grained
+checks can be achieved using bitwise combinations of the other options.
+
+**Options**:
+
+Options can be specified by providing one or more of the following bitwise
+flags:
+
+TSK_CHECK_EDGE_ORDERING
+    Check edge ordering constraints for a tree sequence.
+TSK_CHECK_SITE_ORDERING
+    Check that sites are in nondecreasing position order.
+TSK_CHECK_SITE_DUPLICATES
+    Check for any duplicate site positions.
+TSK_CHECK_MUTATION_ORDERING
+    Check contraints on the ordering of mutations. Any non-null
+    mutation parents and known times are checked for ordering
+    constraints.
+TSK_CHECK_INDEXES
+    Check that the table indexes exist, and contain valid edge
+    references.
+TSK_CHECK_ALL
+    All checks needed to define a valid tree sequence that
+    can be determined without building the individual
+    trees.
```

Comment:
Might be nice to list out the flags this turns on ?

Reply:
Maybe, but I was thinking that this might get out of date as we add more. Anyway, it's all the TSK_CHECK* above, so hopefully that'll be clear.

Reply:
"Defensive documentation" :smile: You're prob right the marginal utility of listing is outweighed by the probability of us forgetting to change it.